src/main/java/com/ss/training/lms/service/BorrowerService.java

```diff
@@ -146,16 +178,15 @@ public boolean checkOutABook(Integer bookId, Integer branchId, Integer cardNo) t
             BookLoan loan = new BookLoan(bookId, branchId, cardNo, now, weekFromNowTS, null);
             
             bookLoanDAO.addBookLoan(loan, conn);
+            success=true;
             conn.commit();
```

Comment:
This line isn't needed since the finally block is handling the commit. If the operation succeeds then conn.commit() will be called twice; as far as I know that won't cause any problems though.